### PR TITLE
Weekly update logic for worlds

### PR DESCRIPTION
**Pull request by Combustible**

### CODE REVIEW

**Review by NickNackGus — COMMENTED** [review]
I'll give this another look once I'm back at my computer.

File: utility_code/weekly_update.py [hunk 1]
```diff
@@ -365,50 +320,80 @@ def err_func(ex, args):
...
 
+            print(f"  {server} - Instances preserved this week: [{','.join(str(x) for x in preserve_instances['dungeon_scores'])}]")
+            for instance in preserve_instances["dungeon_scores"]:
+                world_name = f"{server}{instance}"
+                from_world_path = os.path.join(config["build_path"], world_name)
+                output_world_path = os.path.join(config["output_path"], world_name)
+                if os.path.exists(from_world_path):
+                    move_folder(from_world_path, output_world_path)
+                else:
+                    eprint(f"WARNING: Unable to preserve {world_name} - previous world folder does not exist")
```
[the remark this answers is not in the document]
Comment (Combustible) [reply utility_code/weekly_update.py:368]: Yeah, it does show up. The bot prints stderr results from scripts by default

File: utility_code/lib_py3/common.py [hunk 2]
```diff
@@ -250,18 +265,31 @@ def copy_maps(old, new):
...
         copy_folder(os.path.join(old, path), os.path.join(new, path))
     else:
         copy_file(os.path.join(old, path), os.path.join(new, path))
 
+def move_paths(old, new, paths):
+    for path in paths:
+        try:
+            move_path(old, new, path);
+        except:
+            eprint("*** " + path + " could not be copied, may not exist.")
```
[the remark this answers is not in the document]
Comment (Combustible) [reply utility_code/lib_py3/common.py:285]: Fixed

File: utility_code/weekly_update.py [hunk 3]
```diff
@@ -365,50 +320,80 @@ def err_func(ex, args):
         pprint(last_dict)
         print("Setting rboard scores for $Instances:")
         pprint(instances_dict)
-        rboard.setmulti("$Last", last_dict)
-        rboard.setmulti("$Instances", instances_dict)
+        if len(last_dict) > 0:
+            rboard.setmulti("$Last", last_dict)
+        if len(instances_dict) > 0:
+            rboard.setmulti("$Instances", instances_dict)
```
[the remark this answers is not in the document]
Comment (Combustible) [reply utility_code/weekly_update.py:326]: Good call, but I'm going to remove these in a follow-on commit.

[the remark this answers is not in the document]
Comment (Combustible) [reply utility_code/weekly_update.py:326]: #58

**Pull request merged**